libphobos/m4/autoconf.m4

```diff
@@ -0,0 +1,134 @@
+#
+# Minimal autoconf support for the D language.
+# Adapted from the Go language support files.
+#
+
+# ------------------- #
+# Language selection.
+# ------------------- #
+
+# AC_LANG(D)
+# -----------
+# (we have to use GDC as variable prefix as our GCC patches set GDC
+#  GDC_FOR_BUILD etc. If we ever want to support other D compilers all
+#  names need to be changed to DC)
+AC_LANG_DEFINE([D], [d], [GDC], [],
+[ac_ext=d
+ac_compile='$GDC -c $DFLAGS conftest.$ac_ext >&AS_MESSAGE_LOG_FD'
+ac_link='$GDC -o conftest$ac_exeext $DFLAGS $LDFLAGS conftest.$ac_ext $LIBS >&AS_MESSAGE_LOG_FD'
+ac_compiler_gnu=yes
+])
+
+# AC_LANG_D
+# ----------
+AU_DEFUN([AC_LANG_D], [AC_LANG(D)])
+
+# ------------------- #
+# Producing programs.
+# ------------------- #
+
+# AC_LANG_PROGRAM(D)([PROLOGUE], [BODY])
+# ---------------------------------------
+m4_define([AC_LANG_PROGRAM(D)],
+[module mod;
+$1
+
+int main() {
+  $2
+}])
+
+# _AC_LANG_IO_PROGRAM(D)
+# -----------------------
+# Produce source that performs I/O.
+m4_define([_AC_LANG_IO_PROGRAM(D)],
+[AC_LANG_PROGRAM([import core.stdc.stdio;],
+[FILE *f = fopen ("conftest.out", "w");
+ return ferror (f) || fclose (f) != 0;
+])])
+
+# AC_LANG_CALL(D)(PROLOGUE, FUNCTION)
+# ------------------------------------
+# TODO: Avoid conflicting decl of main?
+m4_define([AC_LANG_CALL(D)],
+[AC_LANG_PROGRAM([$1], [$2(); return 0;])])
+
+# AC_LANG_FUNC_LINK_TRY(D)(FUNCTION)
+# -----------------------------------
+# Try to link a program which calls FUNCTION.
+# This only works for extern(C) functions.
+m4_define([AC_LANG_FUNC_LINK_TRY(D)],
+[AC_LANG_PROGRAM([extern(C) int $1();], [return $1();])])
+
+# AC_LANG_BOOL_COMPILE_TRY(D)(PROLOGUE, EXPRESSION)
+# --------------------------------------------------
+# Return a program which is valid if EXPRESSION is nonzero.
+# Probably not that useful for D, we can extract any information
+# we need using CTFE.
+m4_define([AC_LANG_BOOL_COMPILE_TRY(D)],
+[AC_LANG_PROGRAM([$1],
+[static assert($2); return 0;])])
```

Comment:
These macros are modelled after some standard macros. This implementation is not really useful for D, but I think implementing slightly different behaviour with the same name used by other languages could cause confusion.